src/instructlab/data/list.py

```diff
@@ -0,0 +1,67 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+from pathlib import Path
+from typing import List, TypedDict
+import logging
+import time
+
+# Third Party
+import click
+
+# First Party
+from instructlab import clickext
+from instructlab.configuration import DEFAULTS
+from instructlab.utils import print_table, convert_bytes_to_proper_mag
+
+
+logger = logging.getLogger(__name__)
+
+
+class DatasetListing(TypedDict):
+    """
+    Represents a single dataset listing.
+    """
+
+    filename: str
+    created_at: str
+    file_size: str
+
+
+@click.command(name="list")
+@clickext.display_params
+@click.option(
+    "--dataset-dirs",
+    help="Base directories where datasets are stored.",
+    multiple=True,
+    default=[DEFAULTS.DATASETS_DIR],
+    show_default=True,
+)
+@click.pass_context
+def list_data(ctx, dataset_dirs: list[str]):
+    """lists datasets"""
+    data: List[List[str]] = []
+    for directory in dataset_dirs:
+        dirpath = Path(directory)
+        if not dirpath.exists():
+            ctx.fail(f"directory does not exist: {dirpath}")
+        for entry in Path(directory).iterdir():
```

Comment:
nit: you already have Path (as `dirpath`), no need to instantiate it again.

Reply:
Good catch, I'll remove this.